relay-filter/src/config.rs

```diff
@@ -47,14 +47,9 @@ pub enum LegacyBrowser {
     Unknown(String),
 }
 
-impl<'de> Deserialize<'de> for LegacyBrowser {
-    fn deserialize<D>(deserializer: D) -> Result<Self, D::Error>
-    where
-        D: serde::de::Deserializer<'de>,
-    {
-        let string = Cow::<str>::deserialize(deserializer)?;
-
-        Ok(match string.as_ref() {
+impl From<&str> for LegacyBrowser {
```

Comment:
Please implement `FromStr`, instead.

Reply:
Are you sure? 
I thought that `FromStr` is used for parsing that can fail. 
Since we convert any string into a `LegacyBrowser` ( either a known one or a `Unknown(String)` ) having to deal with a `Result<LegacyBrowser, Self::Err>` seems quite un-ergonomic for no apparent benefit.

Reply:
Yeah, this is an interesting case since we express failure to parse by returning `Unknown`. The standard library uses `Infallible` as error in such cases, for example in [`impl FromStr for String`](https://doc.rust-lang.org/stable/std/str/trait.FromStr.html#impl-FromStr-36).

Otherwise, I'd prefer an explicit infallible `parse` method instead of the conversion trait.

Reply:
ok, done